### ISSUE TITLE
MaxListenerExceededWarning with restify 8.6.0

### ISSUE THREAD

**Issue by luddd3** [restify/node-restify#1883]
When I updated to latest version of Restify. I started to get memory leak warnings.

- [X] Used appropriate template for the issue type
- [X] Searched both open and closed issues for duplicates of this issue
- [X] Title adequately and _concisely_ reflects the feature or the bug

**Restify Version**: 8.6.0
**Node.js Version**: 14.18.1

## Expected behaviour
No warnings

## Actual behaviour
```bash
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
    at _addListener (events.js:475:17)
    at Socket.addListener (events.js:497:10)
    at Socket.Readable.on (internal/streams/readable.js:853:35)
    at Socket.socketListenerWrap [as on] (_http_server.js:950:54)
    at Socket.once (events.js:541:8)
    at readBody (/code/node_modules/restify/lib/plugins/bodyReader.js:189:20)
    at nextTick (/code/node_modules/restify/lib/chain.js:167:13)
    at processTicksAndRejections (internal/process/task_queues.js:77:11)
```

## Repro case
I don't know how to minimally reproduce it. I am running with a fairly large server in prod.

## Cause
I suspect that this is the cause: https://github.com/restify/node-restify/pull/1880

## Are you willing and able to fix this?
No, I am not able to fix it beyond just reverting the change in the pull request referenced above.

**Comment by pcwiek** [comment]
We noticed the same thing. Bumping up the listener limit to 15 doesn't help either, then we're just getting `(...) 16 close listeners added to [Socket] (..)`

**Comment by svedbg** [comment]
Same issue observed in our organization and definitely the root cause is https://github.com/restify/node-restify/pull/1880 
The following line introduces the leak: https://github.com/restify/node-restify/pull/1880/files#diff-556d8ee937e05276d90e3095ea5cd06908d128e446aac53fe57394896569585aR189
If I switch back from:
`req.socket.once('close', next);`
to
`req.once('close', next);`
the leak is not observable.

**Comment by mvoorberg** [comment]
Same issue on node v14.15.4 when using 8.6.0. Pinning at 8.5.1 for now.

**Comment by svedbg** [comment]
Any update on this one?

**Comment by josephharrington** [comment]
This should be fixed in v8.6.1 released today.
